boards/arc/nsim_em/arc_mpu_regions.c

```diff
@@ -11,22 +11,6 @@
 
 #ifdef CONFIG_USERSPACE
 static struct arc_mpu_region mpu_regions[] = {
-#if CONFIG_ARC_MPU_VER == 3 && defined(CONFIG_APPLICATION_MEMORY)
-	/* Region ICCM */
-	MPU_REGION_ENTRY("IMAGE ROM",
-			 (u32_t) _image_rom_start,
-			 (u32_t) _image_rom_size,
-			 REGION_FLASH_ATTR),
-	MPU_REGION_ENTRY("APP MEMORY",
-			 (u32_t) __app_ram_start,
-			 (u32_t) __app_ram_size,
-			 REGION_RAM_ATTR),
-	MPU_REGION_ENTRY("KERNEL MEMORY",
-			 (u32_t) __kernel_ram_start,
```

Comment:
For V3 MPU overlapping regions are not allowed. Explicit, static access should be given to __kernel_ram, other memory access is handled via dynamic mpu programming. This kernel region should given access, instead of the full DCCM as per below code now active without the #ifdef. Giving access to full DCCM results in overlapping regions.

Reply:
I'm confused.
This code is behind the CONFIG_APPLICATION_MEMORY ifdef.
It sounds like the ARC MPU driver is broken and this is a problem unrelated to this PR. Memory domains / app shared memory should work without being dependent on CONFIG_APPLICATION_MEMORY.
Can you fix this in another PR?

Reply:
Think this is addressed now after offline discussion

Reply:
Yes, it is fine now for nsim_sem - remaining assumption I believe is that CONFIG_USERSPACE implies use of memory domains, but that's ok.

Please do the same change for EMSK (boards/arc/em_starterkit/arc_mpu_regions.c)...

Reply:
> remaining assumption I believe is that CONFIG_USERSPACE implies use of memory domains

This is a hard requirement

> Please do the same change for EMSK (boards/arc/em_starterkit/arc_mpu_regions.c)...

I'd really rather not, since I have no way of testing it. This seems an issue which pre-dates my patch, can your team do this in a separate PR?

Reply:
Fair, we'll do.